testsuite/MDAnalysisTests/coordinates/test_openmm.py

```diff
@@ -0,0 +1,241 @@
+
+    def setUp(self):
+        self.universe = mda.Universe(app.PDBFile(RefAdKSmall.filename))
+        self.ref = mda.Universe(RefAdKSmall.filename)
+        self.prec = 3
+
+    def test_dimensions(self):
+        assert_almost_equal(
+            # Angles seem to parse differently when openmm reads the pdb file
+            self.universe.trajectory.ts.dimensions[0:3],
+            self.ref.trajectory.ts.dimensions[0:3],
+            self.prec,
+            "OpenMMPDBFileReader failed to get unitcell dimensions " +
+            "from OpenMMPDBFile",
+        )
+
+    def test_coordinates(self):
+        up = self.universe.atoms.positions
+        rp = self.ref.atoms.positions
+        assert_almost_equal(up, rp, decimal=3)
+
+
+class TestOpenMMModellerReader(_SingleFrameReader):
+    __test__ = True
+
+    def setUp(self):
+        pdb_obj = app.PDBFile(RefAdKSmall.filename)
+        modeller = app.Modeller(pdb_obj.topology, pdb_obj.positions)
+        self.universe = mda.Universe(modeller)
+        self.ref = mda.Universe(RefAdKSmall.filename)
+        self.prec = 3
+
+    def test_dimensions(self):
+        assert_almost_equal(
+            # Angles seem to parse differently when openmm reads the pdb file
+            self.universe.trajectory.ts.dimensions[0:3],
+            self.ref.trajectory.ts.dimensions[0:3],
+            self.prec,
+            "OpenMMModellerReader failed to get unitcell dimensions " +
+            "from OpenMMModeller",
+        )
+
+    def test_coordinates(self):
+        up = self.universe.atoms.positions
+        rp = self.ref.atoms.positions
+        assert_almost_equal(up, rp, decimal=3)
+
+
+class TestOpenMMSimulationReader(_SingleFrameReader):
+    __test__ = True
+
+    def setUp(self):
+        pdb = app.PDBFile(RefAdKSmall.filename)
+        forcefield = app.ForceField("amber99sbildn.xml")
+        system = forcefield.createSystem(
+            pdb.topology, nonbondedMethod=app.NoCutoff, constraints=app.HBonds
+        )
+        integrator = mm.LangevinIntegrator(
+            300 * unit.kelvin, 1 / unit.picoseconds, 2.0 * unit.femtoseconds
+        )
+        sim = app.Simulation(pdb.topology, system, integrator)
+        sim.context.setPositions(pdb.positions)
+        self.universe = mda.Universe(sim)
+        self.ref = mda.Universe(RefAdKSmall.filename)
+        self.prec = 3
+
+    def test_dimensions(self):
+        assert_almost_equal(
+            # Angles seem to parse differently when openmm reads the pdb file
+            self.universe.trajectory.ts.dimensions[0:3],
+            self.ref.trajectory.ts.dimensions[0:3],
+            self.prec,
+            "OpenMMSimulationReader failed to get unitcell dimensions " +
+            "from OpenMMSimulation",
+        )
+
+    def test_coordinates(self):
+        up = self.universe.atoms.positions
+        rp = self.ref.atoms.positions
+        assert_almost_equal(up, rp, decimal=3)
+
+    def test_pickle_singleframe_reader(self):
```

Comment:
Was this going to be a test?

Reply:
Since the underlying openmm objects are c-objects translated with swig (or something along those lines), there were issues trying to pickle these openmm python objects. To be honest, I don't know what's the best way to handle the pickle test, so I just ignored it with `pass`. And if I don't overwrite this test, I think it defaults to the basetest, which will try to pickle the reader, leading to `TypeError: can't pickle SwigPyObject objets`

Reply:
@orbeckst @yuxuanzhuang should we note this as an exception then?

Reply:
Generally speaking, we have not started to do a thorough test for the serialization of converters (#2887), so it largely depends on the upstream support. 

From my very quick test on the picklability of the `openmm.Simulation` object, it seems that most of the data structures can be serialized except `('context', <simtk.openmm.openmm.Context; proxy of <Swig Object of type 'OpenMM::Context *'`

It seems that the openmm people only test the picklability of the `state` obtained from the `Context` https://github.com/openmm/openmm/blob/be19e0222ddf66f612016a3c1f687161a53c2396/wrappers/python/tests/TestPickle.py#L65. As I have no experience in openmm, I cannot give a better solution other than noting this as an exception (and commenting on to this PR #2887).

Reply:
Quick drive-by comment (sorry if off-topic): Please do not have tests just `pass` — that implies that we have functionality that's not present. Instead decorate it with an xfail (unless there's a better solution).

Reply:
Thank you @yuxuanzhuang for looking into it. @ahy3nz if you could change the test to xfail with this reasoning and maybe a link to #2887 or @yuxuanzhuang's comment, that would be great!